### PR TITLE
Potential fix for bad data in billing

### LINKED ISSUE #3571: Billing Poller seems inaccurate.
The Billing Poller doesn't seem to be corresponding to the actual traffic on the interface, The quick graphs bear no relation to the supposedly accurate graphs.

Between the Period of 02:00 and 04:00 where the billing poller seems to be claiming 25GB per hour the actual PPPoE session to the ISP was down due to a fault within the carriers network.

r1
r2
r3

There are also several days where libreNMS will claim 10's of GB usage where as my provider is showing much less usage.

Version	47c22f6
DB Schema	#115
Apache	2.4.10 (Debian)
PHP	5.6.20-0+deb8u1
MySQL	10.0.23-MariaDB-0+deb8u1
RRDtool	1.4.8

### PR DESCRIPTION
**Pull request by laf**
Sanity check the in/out_delta value, if it's out of the range of the ifSpeed then we use the previous value.

Fix #3571

### CODE REVIEW

File: html/billing-graph.php [hunk 1]
```diff
@@ -131,8 +131,8 @@
     $period    = $row['period'];
     $in_delta  = $row['in_delta'];
     $out_delta = $row['out_delta'];
-    $in_value  = round(($in_delta * 8 / $period), 2);
-    $out_value = round(($out_delta * 8 / $period), 2);
+    $in_value  = delta_to_bits($in_delta, $period);
+    $out_value = delta_to_bits($out_delta, $period);;
```
Comment (f0o) [inline html/billing-graph.php]: There's a semicolon too much:)

**Comment by f0o** [comment]
I'm not happy with it as i see an issue with interfaces that have wrong ifSpeed set (for me: tap* are shown as 10mbps when they push several 100mbps).

Could we add a config switch to optionally disable this fix?

**Comment by laf** [comment]
I did think about that, seems like the ideal time for us to support altering the ifSpeed manually.

**Comment by f0o** [comment]
Agreed

**Comment by laf** [comment]
ifSpeed override has now been merged so this should be good.

**Comment by laf** [comment]
bump

**Comment by laf** [comment]
bump

**Comment by laf** [comment]
bump

**Comment by laf** [comment]
bump

**Comment by Rosiak** [comment]
@librenms/reviewers Anyone who got billing running who would be able to validate this?

**Comment by murrant** [comment]
I don't know anything about billing.

**Comment by laf** [comment]
Ok this should be all good to go.

This now resolves another issue which was where two bills had the same port being monitored, the first would log real data but the older one wouldn't as getLastPortCounter would only get the info based on port and not bill+port.

**Pull request merged**